nussl/separation/duet.py

```diff
@@ -0,0 +1,440 @@
+        num_sources (int): Number of sources to find.
+        attenuation_min (int): Lower bound on attenuation value, used as minimum distance in utils.find_peak_indices.
+        attenuation_max (int): Upper bound on attenuation, used for creating a histogram without outliers.
+        num_attenuation_bins (int): Number of bins for attenuation. 
+        delay_min (int): Lower bound on delay, used as minimum distance in utils.find_peak_indices.
+        delay_max (int): Upper bound on delay, used for creating a histogram without outliers.
+        num_delay_bins (int): Number of bins for delay. 
+        peak_threshold (float): Value in [0, 1] for peak picking. 
+        attenuation_min_distance (int): Minimum distance between peaks wrt attenuation. 
+        delay_min_distance (int): Minimum distance between peaks wrt delay.
+        p (int): Weight the histogram with the symmetric attenuation estimator
+        q (int): Weight the histogram with the delay estimator
+
+
+    Attributes:
+        stft_ch0 (np.array): A Numpy matrix containing the stft data of channel 0.
+        stft_ch1 (np.array): A Numpy matrix containing the stft data of channel 0.
+        frequency_matrix (np.array): A Numpy matrix containing the frequencies of analysis.
+        symmetric_atn (np.array): A Numpy matrix containing the symmetric attenuation between the two channels.
+        delay (np.array): A Numpy matrix containing the delay between the two channels.
+        num_time_bins (np.array): The number of time bins for the frequency matrix and mask arrays.
+        num_frequency_bins (int): The number of frequency bins for the mask arrays.
+        attenuation_bins (int): A Numpy array containing the attenuation bins for the histogram.
+        delay_bins (np.array): A Numpy array containing the delay bins for the histogram.
+        normalized_attenuation_delay_histogram (np.array): A normalized Numpy matrix containing the
+                                                           attenuation delay histogram, which has peaks for each source.
+        attenuation_delay_histogram (np.array): A non-normalized Numpy matrix containing the
+                                                attenuation delay histogram, which has peaks for each source.
+        peak_indices (np.array): A Numpy array containing the indices of the peaks for the histogram.
+        separated_sources (np.array): A Numpy array of arrays containing each separated source.
+
+    Examples:
+        :ref:`The DUET Demo Example <duet_demo>`
+
+    """
+
+    def __init__(self, input_audio_signal, num_sources,
+                 attenuation_min=-3, attenuation_max=3, num_attenuation_bins=50,
+                 delay_min=-3, delay_max=3, num_delay_bins=50,
+                 peak_threshold=0.2, attenuation_min_distance=5, delay_min_distance=5, p=1, q=0,
+                 mask_type=mask_separation_base.MaskSeparationBase.BINARY_MASK, mask_threshold=0.5):
```

Comment:
Don't expose `mask_type` and `mask_threshold` because DUET only produces binary masks. So get rid of these as parameters to `__init__()` and when you call the constructor of the super class (in line 74 below), just set `mask_type=BINARY_MASK` (or whatever the syntax) and no need to set anything for `mask_threshold` just leave it out.